ui/app/contexts/metametrics.js

```diff
@@ -1,24 +1,92 @@
-import { Component } from 'react'
-import { connect } from 'react-redux'
+import React, { Component, createContext, useEffect, useRef, useCallback } from 'react'
+import { connect, useSelector } from 'react-redux'
 import PropTypes from 'prop-types'
-import { withRouter } from 'react-router-dom'
+import { withRouter, useLocation } from 'react-router-dom'
 import { compose } from 'redux'
 import {
   getCurrentNetworkId,
   getSelectedAsset,
   getAccountType,
   getNumberOfAccounts,
   getNumberOfTokens,
+} from '../selectors/selectors'
+import {
   txDataSelector,
-} from '../../../selectors'
-
-import { getEnvironmentType } from '../../../../../app/scripts/lib/util'
+} from '../selectors/confirm-transaction'
+import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import {
   sendMetaMetricsEvent,
   sendCountIsTrackable,
-} from '../../utils/metametrics.util'
+} from '../helpers/utils/metametrics.util'
+
+export const MetaMetricsContext = createContext(() => {})
```

Comment:
We should probably log something here if for some reason some downstream usage of the context API hits a fiber that doesn't have an active `<MetaMetricsProvider>`

Reply:
@MetaMask/devs thoughts on whether we should log to sentry (?) or something some value here as a smoke signal? If yes, what should that thing be?

Reply:
Logging to Sentry is a good idea. I'd suggest just throwing an Error - that should get caught by Sentry automatically.

Reply:
Actually, it might be better to call `captureException` explicitly and pass it the error, because the Error might end up being caught and handled somewhere as if it was a failed network request.